mlx/primitives.cpp

```diff
@@ -2926,6 +2927,68 @@ std::vector<array> Scatter::jvp(
   throw std::runtime_error("[scatter] JVP not yet implemented");
 }
 
+std::pair<std::vector<array>, std::vector<int>> Scatter::vmap(
+    const std::vector<array>& inputs,
+    const std::vector<int>& vmap_axes) {
+  assert(inputs.size() >= 2);
+  assert(inputs.size() == vmap_axes.size());
+
+  for (size_t i = 1; i < vmap_axes.size(); i++) {
+    if (vmap_axes[i] != -1) {
+      throw std::runtime_error(
+          "[scatter]: vmap is only supported on the src input.");
```

Comment:
I'm curious about this restriction? Why not allow the update or the indices to be vmapped?

Reply:
For `updates`: If we take one axis of `updates` as vmap dimension, it means scattering the remaining tensor and overwriting the same locations of `src` multiple times (as many as `updates.shape[updates_ax]`). It didn't seem very useful to me, plus we would have to define the semantics (the "last" update wins"?).

`indices` doesn't have tensor semantics, and it's also matched to `axes` (which can't have a vmap dimension because it's just a `vector<int>`. I struggled to imagine how to make the resulting operation clear. I guess we could slice off the `vmap` dimension off `indices`, then check that `axes` covers the remaining `indices`.